src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/ContentService.cs

```diff
@@ -389,94 +385,59 @@ public class ContentService : IContentService
 
-                    var newComment = _context.Comment.Add(new Comment
+                    var comment = new Comment
                     {
                         Id = new Guid(),
-                        IContentBlockId = contentBlockId,
-                        Name = comment.Name,
-                        State = EnumUtil.GetFromString<CommentState>(comment.State),
-                        Time = comment.Time,
-                        CommentText = comment.CommentText,
-                        ResolvedBy = comment.ResolvedBy,
-                        ResolvedOn = comment.ResolvedOn,
-                    });
-
+                        ContentBlockId = contentBlockId,
+                        Content = request.Content,
+                        Created = DateTime.UtcNow,
+                        CreatedById = _userService.GetUserId()
+                    };
+                    
+                    await _context.Comment.AddAsync(comment);
                     await _context.SaveChangesAsync();
-                    return _mapper.Map<CommentViewModel>(newComment.Entity);
+
+                    var added = await _context.Comment
+                        .Include(c => c.CreatedBy)
+                        .FirstAsync(c => c.Id == comment.Id);
+                    return _mapper.Map<CommentViewModel>(added);
                 }
             );
         }
 
-        public Task<Either<ActionResult, CommentViewModel>> UpdateCommentAsync(Guid releaseId,
-            Guid contentSectionId, Guid contentBlockId, Guid commentId,
-            UpdateCommentRequest commentRequest)
+        public Task<Either<ActionResult, CommentViewModel>> UpdateCommentAsync(Guid commentId,
+            AddOrUpdateCommentRequest request)
         {
-            return CheckContentSectionExists(releaseId, contentSectionId)
-                .OnSuccess(CheckCanUpdateRelease)
-                .OnSuccess(async tuple =>
-                {
-                    var (_, section) = tuple;
-
-                    var contentBlock = section.Content.Find(block => block.Id == contentBlockId);
-
-                    if (contentBlock == null)
+            return _persistenceHelper.CheckEntityExists<Comment>(commentId)
+                .OnSuccess(_userService.CheckCanUpdateComment)
+                .OnSuccess(async comment =>
                     {
-                        return ValidationActionResult<CommentViewModel>(ContentBlockNotFound);
-                    }
-
-                    var comment = contentBlock.Comments.Find( c => c.Id == commentId);
+                        _context.Comment.Update(comment);
+                        comment.Content = request.Content;
+                        comment.Updated = DateTime.UtcNow;
+                        await _context.SaveChangesAsync();
 
-                    if (comment == null)
-                    {
-                        return ValidationActionResult(CommentNotFound);
+                        var updated = await _context.Comment
```

Comment:
Is there not a "GetCommentById" method we can call here already?

Reply:
As above, I've added it now to make this code reusable.